tests/bofire/strategies/doe/test_design.py

```diff
@@ -550,3 +550,122 @@ def test_partially_fixed_experiments():
         check_partially_and_fully_fixed_experiments(
             domain, 2, fixed_experiments, partially_fixed_experiments
         )
+
+    fixed_experiments = pd.DataFrame(
+        np.array([[1, 0, 0, 0], [0, 1, 0.7, 1]]), columns=domain.inputs.get_keys()
+    )
+
+    doe = find_local_max_ipopt(
+        domain, "linear", n_experiments=3, fixed_experiments=fixed_experiments
+    ).reset_index(drop=True)
+
+    assert doe.shape == (3, 4)
+    assert np.allclose(doe.iloc[[0, 1]]["x1"], fixed_experiments["x1"])
+    assert np.allclose(doe.iloc[[0, 1]]["x2"], fixed_experiments["x2"])
+    assert np.allclose(doe.iloc[[0, 1]]["a1"], fixed_experiments["a1"])
+    assert np.allclose(doe.iloc[[0, 1]]["a2"], fixed_experiments["a2"])
+
+    fixed_experiments = pd.DataFrame(
+        np.array([[1, 0, 0], [0, 1, 0.7]]), columns=domain.inputs.get_keys()[:-1]
+    )
+
+    with pytest.raises(ValueError):
+        doe = find_local_max_ipopt(
+            domain, "linear", n_experiments=2, fixed_experiments=fixed_experiments
+        )
+
+    fixed_experiments = pd.DataFrame(
+        np.array([[1, 0, 0, 0, 1], [0, 1, 0.7, 1, 2]]),
+        columns=domain.inputs.get_keys() + ["c0"],
+    )
+
+    doe = find_local_max_ipopt(
+        domain, "linear", n_experiments=3, fixed_experiments=fixed_experiments
+    ).reset_index(drop=True)
+
+    assert doe.shape == (3, 4)
+    assert np.allclose(doe.iloc[[0, 1]]["x1"], fixed_experiments["x1"])
+    assert np.allclose(doe.iloc[[0, 1]]["x2"], fixed_experiments["x2"])
+    assert np.allclose(doe.iloc[[0, 1]]["a1"], fixed_experiments["a1"])
+    assert np.allclose(doe.iloc[[0, 1]]["a2"], fixed_experiments["a2"])
+
+    partially_fixed_experiments = pd.DataFrame(
+        np.array([[1.0, None, None, None, 1.0], [0.0, None, None, None, 2.0]]),
+        columns=["x1", "x2", "a1", "a2", "c0"],
+    )
+    doe = find_local_max_ipopt(
+        domain,
+        "linear",
+        n_experiments=3,
+        partially_fixed_experiments=partially_fixed_experiments,
+    ).reset_index(drop=True)
+
+    assert doe.shape == (3, 4)
+    assert np.allclose(
+        doe.iloc[[0, 1]]["x1"], partially_fixed_experiments["x1"].astype(float)
+    )
+
+    doe = find_local_max_ipopt(
+        domain,
+        "linear",
+        n_experiments=4,
+        fixed_experiments=fixed_experiments,
+        partially_fixed_experiments=partially_fixed_experiments,
+    ).reset_index(drop=True)
+
+    assert doe.shape == (4, 4)
+    assert np.allclose(doe.iloc[[0, 1]]["x1"], fixed_experiments["x1"])
+    assert np.allclose(doe.iloc[[0, 1]]["x2"], fixed_experiments["x2"])
+    assert np.allclose(doe.iloc[[0, 1]]["a1"], fixed_experiments["a1"])
+    assert np.allclose(doe.iloc[[0, 1]]["a2"], fixed_experiments["a2"])
+    assert np.allclose(
+        doe.iloc[[2, 3]]["x1"], partially_fixed_experiments["x1"].astype(float)
+    )
+
+    with pytest.raises(ValueError):
```

Comment:
Ideally, we would also catch here the exact error message to be sure that we are catching the correct stuff.